signer/jws_test.go

```diff
@@ -0,0 +1,148 @@
+package signer
+
+import (
+	"crypto/x509"
+	"errors"
+	"github.com/notaryproject/notation-core-go/internal/testhelper"
+	"testing"
+)
+
+func TestNewJWSEnvelopeFromBytes(t *testing.T) {
+	t.Run("newJWSEnvelopeFromBytes", func(t *testing.T) {
+		if _, err := newJWSEnvelopeFromBytes([]byte(TestValidSig)); err != nil {
+			t.Errorf("Error found")
+		}
+	})
+
+	t.Run("newJWSEnvelopeFromBytes Error", func(t *testing.T) {
+		if _, err := newJWSEnvelopeFromBytes([]byte("Malformed")); err == nil {
+			t.Errorf("Expected error but not found")
+		}
+	})
+}
+
+func TestValidateIntegrity(t *testing.T) {
+	t.Run("with newJWSEnvelope() returns error", func(t *testing.T) {
+		env := jwsEnvelope{}
+		err := env.validateIntegrity()
+		if !(err != nil && errors.As(err, new(SignatureNotFoundError))) {
+			t.Errorf("Expected SignatureNotFoundError but found %q", err)
+		}
+	})
+
+	t.Run("with NewJWSEnvelopeFromBytes works", func(t *testing.T) {
+		env, _ := newJWSEnvelopeFromBytes([]byte(TestValidSig))
+		err := env.validateIntegrity()
+		if err != nil {
+			t.Errorf("validateIntegrity(). Error = %s", err)
+		}
+	})
+
+	t.Run("with invalid base64 bytes sig envelope returns error", func(t *testing.T) {
+		env, _ := newJWSEnvelopeFromBytes([]byte("{\"Payload\":\"Hi!\",\"Protected\":\"Hi\",\"Header\":{},\"Signature\":\"Hi!\"}"))
+		if err := env.validateIntegrity(); err == nil {
+			t.Errorf("Expected error but not found")
```

Comment:
Can we also test the type of the expected err?

Reply:
yep, done